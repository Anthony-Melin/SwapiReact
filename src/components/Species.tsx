import ListValueFactory from "./__components__/ListValueFactory.tsx";

export default ListValueFactory('species', 'Species', 'species')
