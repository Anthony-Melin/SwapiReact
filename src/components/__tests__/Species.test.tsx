import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import Species from '../Species.tsx'

describe('Species', () => {
    describe('with empty list', () => {
        beforeEach(() => {
            render(<Species values={[]} />)
        })

        it('should match snapshot', () => {
            expect(screen.getByTestId('species')).toMatchSnapshot()
        })

        it('should have a title', () => {
            expect(screen.getByTestId('species')).toHaveTextContent('Species')
        })

        it('should have empty list label', () => {
            expect(screen.getByTestId('species-empty')).toHaveTextContent('No species linked')
        })
    })

    describe('with filled list', () => {
        beforeEach(() => {
            render(<Species values={['VALUE_1', 'VALUE_2']} />)
        })

        it('should match snapshot', () => {
            expect(screen.getByTestId('species')).toMatchSnapshot()
        })

        it('should have a title', () => {
            expect(screen.getByTestId('species')).toHaveTextContent('Species')
        })

        it('should contains elements', () => {
            expect(screen.getByTestId('species-0')).toHaveTextContent('VALUE_1')
            expect(screen.getByTestId('species-1')).toHaveTextContent('VALUE_2')
        })
    })
})
