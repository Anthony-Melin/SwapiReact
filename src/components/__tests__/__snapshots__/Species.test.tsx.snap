// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Species > with empty list > should match snapshot 1`] = `
<div
  data-testid="species"
>
  <p
    data-testid="species-title"
  >
    Species
  </p>
  <p
    data-testid="species-empty"
  >
    No 
    species
     linked
  </p>
</div>
`;

exports[`Species > with filled list > should match snapshot 1`] = `
<div
  data-testid="species"
>
  <p
    data-testid="species-title"
  >
    Species
  </p>
  <ul>
    <li
      data-testid="species-0"
    >
      VALUE_1
    </li>
    <li
      data-testid="species-1"
    >
      VALUE_2
    </li>
  </ul>
</div>
`;
